### DISCUSSION TITLE
Import Google Takeout as a folder

### DISCUSSION THREAD

**Discussion by MatMaul** [ente-io/ente#591] [Enhancements]
Hi,

First thank you for this great software 🙂

Could you allow to take a folder as an input for the Google Takeout import ? I would like to maximize my chances of having json metadata alongside the picture files by uncompressing all my zip files to a single folder.

You may want to be able to take a TGZ too, since it is a possibility at export time.

Thanks !

**Comment by vishnukvmd** [comment]
> Could you allow to take a folder as an input for the Google Takeout import 

We already support this.

Merge all your Takeout.zips into a single folder, drag and drop it into our Desktop app, and it will merge the metadata with the respective files.

**Reply by MatMaul** [reply]
Thanks for the fast answer!

I tried the `Google Takeout` button after hitting `upload` and couldn't select a folder, hence my confusion. I'll try that this week, after trying to put a bit of order in the mess that is this "export", (no) thanks Google.

**Comment by sagz** [comment]
Can you point to the code which ingests the Google Takeout Zips? I am curious to know how it compares to metadatafixer.com wrt the JSON->EXIF part.

Also, is the Google Takeout import feature able to handle sidecar files like the movie attachments of Apple Live Photos or Google Motion Photos?

**Reply by mnvr** [reply]
> Can you point to the code which ingests the Google Takeout Zips? I am curious to know how it compares to metadatafixer.com wrt the JSON->EXIF part.

This should be the bit - https://github.com/ente-io/ente/blob/main/web/apps/photos/src/services/upload/metadataService.ts#L153
